### PR TITLE
Adding support for files to the content-plugin

### PR DESCRIPTION
**Pull request by dsimko**
It is possible to add images and attachments to a blog post.

Updated demo: http://46.183.65.66:8080/blog.html

### CODE REVIEW

**Comment by kbachl** [comment]
I'm not sure you really want the URLs made by that... 
e.g.: http://46.183.65.66:8080/content-plugin/bx_ZpWFOgGGBwmZbuHQxLkERLgc/5da7028e-19b5-4014-8bc0-fcf8484d4f0b
or  http://46.183.65.66:8080/content-plugin/bx_ZpWFOgGGBwmZbuHQxLkERLgc/0193ac81-6efa-48d2-b995-a304fbbcefa1

is IMHO nightmare for any seo strategy... also from my wordpress experience one often reuses media files like images over more than 1 blog post... 
beside, I'm not sure that exposing the workspace name/id is a good idea in terms of security (I know that security by obscurity is also a bad idea ;) )

maybe a distinct media workspaceor plugin would be better?
path then could be:

http://127.0.0.1/media-plugin/path/to/file/file.png

where "/media-plugin" could be configured on the plugin level in code...
or you use a reference to the site-workspace and allow to link images from there?


PS: I hope you dont see me as one who is only complaining - I really like you blog/content plugin

File: brix-plugin-content/src/main/java/org/brixcms/plugin/content/ContentPlugin.java [hunk 1]
```diff
@@ -287,4 +290,20 @@ public static String normalizeValue(final String value, int maxLength) {
     protected int getTabPriority() {
         return 1000;
     }
+
+    public void registerNodePlugin(NodeEditorPlugin plugin) {
+        if (plugin == null) {
```
Comment (martin-g) [inline brix-plugin-content/src/main/java/org/brixcms/plugin/content/ContentPlugin.java]: `Args.notNull(plugin, "plugin")` ?

**Comment by dsimko** [comment]
Thank you Korbinian, I've created https://github.com/brix-cms/brix-cms-plugins/issues/4 and I would like to discuss it later.

>PS: I hope you dont see me as one who is only complaining - I really like you blog/content plugin

I really appreciate your feedback!

File: brix-plugin-content/src/main/java/org/brixcms/plugin/content/ContentPlugin.java (hunk 1 above)
Comment (dsimko) [inline brix-plugin-content/src/main/java/org/brixcms/plugin/content/ContentPlugin.java]: Yes, this will be nicer. I am going to change it. Thank you!

**Pull request merged**